diff --git a/arch/arm/dts/imx8ulp-evk.dts b/arch/arm/dts/imx8ulp-evk.dts
index eff4926c36..0f5d2ec76f 100644
--- a/arch/arm/dts/imx8ulp-evk.dts
+++ b/arch/arm/dts/imx8ulp-evk.dts
@@ -225,8 +225,8 @@
 
 	pinctrl_lpuart5: lpuart5grp {
 		fsl,pins = <
-			MX8ULP_PAD_PTF14__LPUART5_TX	0x3
-			MX8ULP_PAD_PTF15__LPUART5_RX	0x3
+			MX8ULP_PAD_PTE6__LPUART5_TX 0x1
+			MX8ULP_PAD_PTE7__LPUART5_RX 0x1
 		>;
 	};
 
@@ -266,6 +266,20 @@
 			MX8ULP_PAD_PTD11__SDHC0_DQS	0x10002
 		>;
 	};
+
+	pinctrl_usdhc2_pte: usdhc2ptegrp {
+		fsl,pins = <
+			MX8ULP_PAD_PTE1__SDHC2_D0	0x3
+			MX8ULP_PAD_PTE0__SDHC2_D1	0x3
+			MX8ULP_PAD_PTE5__SDHC2_D2	0x3
+			MX8ULP_PAD_PTE4__SDHC2_D3	0x3
+			MX8ULP_PAD_PTE2__SDHC2_CLK	0x10002
+			MX8ULP_PAD_PTE3__SDHC2_CMD	0x3
+			MX8ULP_PAD_PTF26__SDHC2_RESET_B	0x3
+			MX8ULP_PAD_PTF27__SDHC2_WP	0x2
+			MX8ULP_PAD_PTF28__SDHC2_CD	0x3
+		>;
+	};
 };
 
 &flexspi0 {
@@ -318,3 +332,19 @@
 &dphy {
 	status = "okay";
 };
+
+&usdhc2 {
+	ssigned-clocks = <&pcc4 IMX8ULP_CLK_USDHC2>;
+	assigned-clock-rates = <396000000>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2_pte>;
+	pinctrl-1 = <&pinctrl_usdhc2_pte>;
+	pinctrl-2 = <&pinctrl_usdhc2_pte>;
+	wp-gpios = <&gpiof 27 GPIO_ACTIVE_HIGH>;
+	cd-gpios = <&gpiof 28 GPIO_ACTIVE_LOW>;
+	bus-width = <4>;
+	keep-power-in-suspend;
+	status = "okay";
+};
+
+
