Observation: ad hoc patch
Upstream-Status: Inappropriate [other]
Signed-off-by: F. Aragon <francisco.aragon@advantech.de>

diff --git a/arch/arm/dts/fsl-imx8qxp-mek-u-boot.dtsi b/arch/arm/dts/fsl-imx8qxp-mek-u-boot.dtsi
index ad460944e7a..cf3eadee9af 100644
--- a/arch/arm/dts/fsl-imx8qxp-mek-u-boot.dtsi
+++ b/arch/arm/dts/fsl-imx8qxp-mek-u-boot.dtsi
@@ -3,6 +3,8 @@
  * Copyright 2018, 2021 NXP
  */
 
+#define SDHC_LOW_SPEED_ONLY
+
 / {
 
 	aliases {
@@ -247,8 +249,12 @@
 
 &usdhc2 {
 	bootph-pre-ram;
+#ifdef SDHC_LOW_SPEED_ONLY
+	no-1-8-v;
+#else
 	sd-uhs-sdr104;
 	sd-uhs-ddr50;
+#endif
 };
 
 &crypto {
@@ -279,10 +285,12 @@
 	compatible = "fsl,imx8qm-lpi2c", "fsl,imx-virt-i2c";
 };
 
+#if 0
 &{/i2c@5a810000/i2cswitch@71} {
 	compatible = "nxp,pca9646", "fsl,imx-virt-i2c-mux";
 	virtual-bus-seq = <12>;
 };
+#endif
 
 &wu {
 	bootph-pre-ram;
diff --git a/arch/arm/dts/fsl-imx8qxp-mek.dts b/arch/arm/dts/fsl-imx8qxp-mek.dts
index 0e955dd3ee2..e1a397800ef 100644
--- a/arch/arm/dts/fsl-imx8qxp-mek.dts
+++ b/arch/arm/dts/fsl-imx8qxp-mek.dts
@@ -7,11 +7,15 @@
 
 #include "fsl-imx8qxp.dtsi"
 
+#define SDHC_LOW_SPEED_ONLY
+
 / {
-	model = "NXP i.MX8QXP MEK";
+	model = "NXP i.MX8QXP ROM-5620 A1/A2";
 	compatible = "fsl,imx8qxp-mek", "fsl,imx8qxp";
 
+#if 0
 	aliases {
+		i2c3 = &i2c0_mipi_lvds1;
 		i2c4 = &bb_i2c1;
 		i2c5 = &mfi_i2c1;
 		i2c6 = &i2cexp1_i2c1;
@@ -19,6 +23,7 @@
 		gpio8 = &pca9557_a;
 		gpio9 = &pca9557_b;
 	};
+#endif
 
 	chosen {
 		bootargs = "console=ttyLP0,115200 earlycon";
@@ -45,7 +50,7 @@
 			regulator-min-microvolt = <3300000>;
 			regulator-max-microvolt = <3300000>;
 			regulator-name = "mpcie_3v3";
-			gpio = <&pca9557_a 2 GPIO_ACTIVE_HIGH>;
+			reset-gpio = <&gpio4 0 GPIO_ACTIVE_LOW>;
 			enable-active-high;
 			regulator-always-on;
 		};
@@ -56,7 +61,7 @@
 			regulator-name = "usb_otg1_vbus";
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
-			gpio = <&pca9557_b 2 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio4 3 GPIO_ACTIVE_LOW>;
 			enable-active-high;
 		};
 
@@ -191,7 +196,7 @@
 				SC_P_QSPI0B_SS1_B_LSIO_QSPI0B_SS1_B	0x06000021
 			>;
 		};
-
+#if 0
 		pinctrl_i2c0_mipi_lvds0: mipi_lvds0_i2c0_grp {
 			fsl,pins = <
 				SC_P_MIPI_DSI0_I2C0_SCL_MIPI_DSI0_I2C0_SCL 0xc6000020
@@ -205,6 +210,7 @@
 				SC_P_MIPI_DSI1_I2C0_SDA_MIPI_DSI1_I2C0_SDA 0xc6000020
 			>;
 		};
+#endif
 	};
 };
 
@@ -254,10 +260,11 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_lpi2c1>;
 	status = "okay";
-
+#if 0
 	i2cswitch@71 {
-		compatible = "nxp,pca9646";
+		compatible = "nxp,pca9646", "fsl,imx-virt-i2c-mux";
 		reg = <0x71>;
+		virtual-bus-seq = <12>;
 		u-boot,i2c-offset-len = <0>;
 		#address-cells = <1>;
 		#size-cells = <0>;
@@ -299,8 +306,10 @@
 			};
 		};
 	};
+#endif
 };
 
+#if 0
 &i2c0_mipi_lvds0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
@@ -332,6 +341,7 @@
 	clock-frequency = <100000>;
 	status = "okay";
 };
+#endif
 
 &usdhc1 {
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
@@ -352,6 +362,9 @@
 	cd-gpios = <&gpio4 22 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio4 21 GPIO_ACTIVE_HIGH>;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
+#ifdef SDHC_LOW_SPEED_ONLY
+	no-1-8-v;
+#endif
 	status = "okay";
 };
 
@@ -372,9 +385,11 @@
 	fsl,ar8031-phy-fixup;
 	fsl,magic-packet;
 	status = "okay";
+#if 0
 	phy-reset-gpios = <&pca9557_a 4 GPIO_ACTIVE_LOW>;
 	phy-reset-duration = <10>;
 	phy-reset-post-delay = <150>;
+#endif
 
 	mdio {
 		#address-cells = <1>;
@@ -419,6 +434,7 @@
 	status = "okay";
 };
 
+#if 0
 &ldb1_phy {
 	status = "okay";
 };
@@ -456,3 +472,4 @@
 		};
 	};
 };
+#endif
diff --git a/board/freescale/imx8qxp_mek/imx8qxp_mek.c b/board/freescale/imx8qxp_mek/imx8qxp_mek.c
index 281e8962b89..54eb8636b3a 100644
--- a/board/freescale/imx8qxp_mek/imx8qxp_mek.c
+++ b/board/freescale/imx8qxp_mek/imx8qxp_mek.c
@@ -63,6 +63,14 @@ int board_early_init_f(void)
 
 	setup_iomux_uart();
 
+/* Dual bootloader feature will require CAAM access, but JR0 and JR1 will be
+ * assigned to seco for imx8, use JR3 instead.
+ */
+#if defined(CONFIG_SPL_BUILD) && defined(CONFIG_DUAL_BOOTLOADER)
+	sc_pm_set_resource_power_mode(-1, SC_R_CAAM_JR3, SC_PM_PW_MODE_ON);
+	sc_pm_set_resource_power_mode(-1, SC_R_CAAM_JR3_OUT, SC_PM_PW_MODE_ON);
+#endif
+
 	return 0;
 }
 
@@ -89,6 +97,136 @@ static void board_gpio_init(void)
 static inline void board_gpio_init(void) {}
 #endif
 
+#if IS_ENABLED(CONFIG_FEC_MXC)
+#include <miiphy.h>
+
+#ifndef CONFIG_DM_ETH
+static iomux_cfg_t pad_enet1[] = {
+	SC_P_SPDIF0_TX | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_SPDIF0_RX | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ESAI0_TX3_RX2 | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ESAI0_TX2_RX3 | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ESAI0_TX1 | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ESAI0_TX0 | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ESAI0_SCKR | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ESAI0_TX4_RX1 | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ESAI0_TX5_RX0 | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ESAI0_FST  | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ESAI0_SCKT | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ESAI0_FSR  | MUX_MODE_ALT(3) | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+
+	/* Shared MDIO */
+	SC_P_ENET0_MDC | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_MDIO | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+};
+
+static iomux_cfg_t pad_enet0[] = {
+	SC_P_ENET0_RGMII_RX_CTL | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ENET0_RGMII_RXD0 | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ENET0_RGMII_RXD1 | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ENET0_RGMII_RXD2 | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ENET0_RGMII_RXD3 | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ENET0_RGMII_RXC | MUX_PAD_CTRL(ENET_INPUT_PAD_CTRL),
+	SC_P_ENET0_RGMII_TX_CTL | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_RGMII_TXD0 | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_RGMII_TXD1 | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_RGMII_TXD2 | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_RGMII_TXD3 | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_RGMII_TXC | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+
+	/* Shared MDIO */
+	SC_P_ENET0_MDC | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+	SC_P_ENET0_MDIO | MUX_PAD_CTRL(ENET_NORMAL_PAD_CTRL),
+};
+
+static void setup_iomux_fec(void)
+{
+	if (0 == CONFIG_FEC_ENET_DEV)
+		imx8_iomux_setup_multiple_pads(pad_enet0, ARRAY_SIZE(pad_enet0));
+	else
+		imx8_iomux_setup_multiple_pads(pad_enet1, ARRAY_SIZE(pad_enet1));
+}
+
+static void enet_device_phy_reset(void)
+{
+	struct gpio_desc desc;
+	int ret;
+
+	/* The BB_PER_RST_B will reset the ENET1 PHY */
+	if (0 == CONFIG_FEC_ENET_DEV) {
+		ret = dm_gpio_lookup_name("gpio@1a_4", &desc);
+		if (ret)
+			return;
+
+		ret = dm_gpio_request(&desc, "enet0_reset");
+		if (ret)
+			return;
+
+		dm_gpio_set_dir_flags(&desc, GPIOD_IS_OUT);
+		dm_gpio_set_value(&desc, 0);
+		udelay(50);
+		dm_gpio_set_value(&desc, 1);
+	}
+
+	/* The board has a long delay for this reset to become stable */
+	mdelay(200);
+}
+
+int board_eth_init(bd_t *bis)
+{
+	int ret;
+	struct power_domain pd;
+
+	printf("[%s] %d\n", __func__, __LINE__);
+
+	/* Reset ENET PHY */
+	enet_device_phy_reset();
+
+	if (CONFIG_FEC_ENET_DEV) {
+		if (!power_domain_lookup_name("conn_enet1", &pd))
+			power_domain_on(&pd);
+	} else {
+		if (!power_domain_lookup_name("conn_enet0", &pd))
+			power_domain_on(&pd);
+	}
+
+	setup_iomux_fec();
+
+	ret = fecmxc_initialize_multi(bis, CONFIG_FEC_ENET_DEV,
+		CONFIG_FEC_MXC_PHYADDR, IMX_FEC_BASE);
+	if (ret)
+		printf("FEC1 MXC: %s:failed\n", __func__);
+
+	return ret;
+}
+#endif
+
+int board_phy_config(struct phy_device *phydev)
+{
+	unsigned short val;
+
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0d04);
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x10, 0xa050);
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x11, 0x0000);
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0000);
+
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0d08);
+        val = phy_read(phydev, MDIO_DEVAD_NONE, 0x11);
+        val |= (0x1 << 8);//enable TX delay
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x11, val);
+
+        val = phy_read(phydev, MDIO_DEVAD_NONE, 0x15);
+        val |= (0x1 << 3);//enable RX delay
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x15, val);
+        phy_write(phydev, MDIO_DEVAD_NONE, 0x1f, 0x0000);
+
+	if (phydev->drv->config)
+		phydev->drv->config(phydev);
+
+	return 0;
+}
+#endif
+
 int checkboard(void)
 {
 #ifdef CONFIG_TARGET_IMX8DX_MEK
@@ -207,6 +345,37 @@ int board_usb_cleanup(int index, enum usb_init_type init)
 }
 #endif
 
+static iomux_cfg_t wdt_trig[] = {
+	SC_P_ADC_IN4 | MUX_MODE_ALT(4) | MUX_PAD_CTRL(GPIO_PAD_CTRL),
+};
+
+static void setup_iomux_wdt(void)
+{
+	int ret;
+	struct gpio_desc desc;
+	struct udevice *dev;
+
+	imx8_iomux_setup_multiple_pads(wdt_trig, ARRAY_SIZE(wdt_trig));
+
+	ret = uclass_get_device_by_seq(UCLASS_GPIO, 1, &dev);
+	if (ret) {
+		printf("%s failed to find GPIO1 device, ret = %d\n", __func__, ret);
+		return;
+	}
+
+	desc.dev = dev;
+	desc.offset = 14;
+
+	ret = dm_gpio_request(&desc, "wdt_trig");
+	if (ret) {
+		printf("%s request wdt_trig failed ret = %d\n", __func__, ret);
+		return;
+	}
+
+	dm_gpio_set_dir_flags(&desc, GPIOD_IS_OUT);
+	dm_gpio_set_value(&desc, 1);
+}
+
 int board_init(void)
 {
 	board_gpio_init();
@@ -224,6 +393,7 @@ int board_init(void)
 	}
 #endif
 
+	setup_iomux_wdt();
 	return 0;
 }
 
@@ -257,6 +427,47 @@ int ft_board_setup(void *blob, struct bd_info *bd)
 }
 #endif
 
+static iomux_cfg_t debug_uart_sel_gpio[] = {
+	SC_P_SAI0_RXD | MUX_MODE_ALT(4) | MUX_PAD_CTRL(GPIO_PAD_CTRL),
+};
+
+static void debug_uart_sel(void)
+{
+	int value = -1;
+	int ret;
+	struct gpio_desc desc;
+	struct udevice *dev;
+
+	imx8_iomux_setup_multiple_pads(debug_uart_sel_gpio, ARRAY_SIZE(debug_uart_sel_gpio));
+
+        ret = uclass_get_device_by_seq(UCLASS_GPIO, 0, &dev);
+        if (ret) {
+                printf("%s failed to find GPIO0 device, ret = %d\n", __func__, ret);
+                return;
+        }
+
+        desc.dev = dev;
+        desc.offset = 27;
+
+        ret = dm_gpio_request(&desc, "debug_uart_sel");
+        if (ret) {
+                printf("%s request debug_uart_sel failed ret = %d\n", __func__, ret);
+                return;
+        }
+
+	dm_gpio_set_dir_flags(&desc, GPIOD_IS_IN);
+
+        value = dm_gpio_get_value(&desc);
+
+	/* High: enable debug log. Low: disable debug log. */
+	if(value == 0)
+	{
+		env_set("console", "disabled");
+		env_set("earlycon", "disabled");
+		env_set("mmcargs", "setenv bootargs console=${console},${baudrate} earlycon=${earlycon},${baudrate} root=${mmcroot}");
+	}
+}
+
 int board_late_init(void)
 {
 	char *fdt_file;
@@ -286,12 +497,12 @@ int board_late_init(void)
 		if (m4_booted)
 			env_set("fdt_file", "imx8dx-mek-rpmsg.dtb");
 		else
-			env_set("fdt_file", "imx8dx-mek.dtb");
+			env_set("fdt_file", CONFIG_DEFAULT_FDT_FILE);
 #else
 		if (m4_booted)
 			env_set("fdt_file", "imx8qxp-mek-rpmsg.dtb");
 		else
-			env_set("fdt_file", "imx8qxp-mek.dtb");
+			env_set("fdt_file", CONFIG_DEFAULT_FDT_FILE);
 #endif
 	}
 
@@ -299,6 +510,7 @@ int board_late_init(void)
 	board_late_mmc_env_init();
 #endif
 
+	debug_uart_sel();
 	return 0;
 }
 
diff --git a/board/freescale/imx8qxp_mek/imximage.cfg b/board/freescale/imx8qxp_mek/imximage.cfg
index 88d6955a9ef..f7f806d982c 100644
--- a/board/freescale/imx8qxp_mek/imximage.cfg
+++ b/board/freescale/imx8qxp_mek/imximage.cfg
@@ -6,8 +6,8 @@
  * and create imx8image boot image
  */
 
-
-BOOT_FROM	sd
+/* Boot from SD, sector size 0x400 */
+BOOT_FROM SD 0x400
 /* SoC type IMX8QX */
 SOC_TYPE IMX8QX
 /* Append seco container image */
diff --git a/configs/imx8qxp_mek_defconfig b/configs/imx8qxp_mek_defconfig
index b478723f679..53a528222e4 100644
--- a/configs/imx8qxp_mek_defconfig
+++ b/configs/imx8qxp_mek_defconfig
@@ -8,11 +8,11 @@ CONFIG_SYS_MEMTEST_END=0xB0000000
 CONFIG_SPL_GPIO=y
 CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
-CONFIG_NR_DRAM_BANKS=4
+CONFIG_NR_DRAM_BANKS=8
 CONFIG_HAS_CUSTOM_SYS_INIT_SP_ADDR=y
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x80200000
 CONFIG_ENV_SIZE=0x2000
-CONFIG_ENV_OFFSET=0x700000
+CONFIG_ENV_OFFSET=0x400000
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="fsl-imx8qxp-mek"
 CONFIG_TARGET_IMX8QXP_MEK=y
@@ -183,7 +183,7 @@ CONFIG_FASTBOOT_UUU_SUPPORT=y
 CONFIG_FASTBOOT_BUF_ADDR=0x82800000
 CONFIG_FASTBOOT_BUF_SIZE=0x40000000
 CONFIG_FASTBOOT_FLASH=y
-CONFIG_FASTBOOT_USB_DEV=1
+CONFIG_FASTBOOT_USB_DEV=0
 
 CONFIG_SYS_I2C_IMX_VIRT_I2C=y
 CONFIG_I2C_MUX_IMX_VIRT=y
