Observation: ad hoc patch
Upstream-Status: Inappropriate [other]
Signed-off-by: F. Aragon <francisco.aragon@advantech.de>

diff --git a/iMX8QX/soc.mak b/iMX8QX/soc.mak
index 4817f32..6926f4f 100755
--- a/iMX8QX/soc.mak
+++ b/iMX8QX/soc.mak
@@ -25,8 +25,10 @@ endif
 TEE = tee.bin
 
 ifeq ($(SOC),iMX8DX)
+KERNEL_DTB ?= imx8dx-mek.dtb
 TEE_LOAD_ADDR ?= 0x96000000
 else ifeq ($(SOC),iMX8QX)
+KERNEL_DTB ?= imx8qxp-mek.dtb
 TEE_LOAD_ADDR ?=  0xFE000000
 endif
 
@@ -145,8 +147,8 @@ flash_regression_linux_m4_xip : $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin
 flash_scfw: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin
 	./$(MKIMG) -soc QX -rev B0 -dcd skip -append $(AHAB_IMG) -c -scfw scfw_tcm.bin -out flash.bin
 
-flash_kernel: $(MKIMG) Image imx8qxp-mek.dtb
-	./$(MKIMG) -soc QX -rev B0 -c -ap Image a35 0x80200000 --data imx8qxp-mek.dtb a35 0x83000000 -out flash.bin
+flash_kernel: $(MKIMG) Image $(KERNEL_DTB)
+	./$(MKIMG) -soc QX -rev B0 -c -ap Image a35 0x80200000 --data $(KERNEL_DTB) a35 0x83000000 -out flash.bin
 
 parse_container: $(MKIMG) flash.bin
 	./$(MKIMG) -soc QX -rev B0 -parse flash.bin
